medusa/processTV.py

```diff
@@ -489,54 +487,55 @@ def unRAR(path, rarFiles, force, result):  # pylint: disable=too-many-branches,t
     :param force: Force checking when already checking (currently unused)
     :param result: True if file is already postprocessed, False if not
     :return:
     """
     if force:
         return False
 
-    # Avoid processing the same dir again if we use a process method <> move
+    file_path = os.path.join(dir_name, video_file)
+    file_size = os.path.getsize(file_path)
+
     main_db_con = db.DBConnection()
-    sql_result = main_db_con.select("SELECT release_name FROM tv_episodes WHERE release_name IN (?, ?) LIMIT 1", [dirName, videofile.rpartition('.')[0]])
-    if sql_result:
-        # result.output += logHelper(u"You're trying to post process a dir that's already been processed, skipping", logger.DEBUG)
+    # Try generic method first
+    episodes_result = main_db_con.select(
+        'SELECT file_size '
+        'FROM tv_episodes '
+        'WHERE release_name LIKE ?',
+        [remove_extension(video_file)])
+
+    if episodes_result and episodes_result[0]['file_size'] == file_size:
+        result.output += logHelper(u'New file has the same name and size, skipping it', logger.DEBUG)
         return True
 
-    # Needed if we have downloaded the same episode @ different quality
-    # But we need to make sure we check the history of the episode we're going to PP, and not others
-    try:  # if it fails to find any info (because we're doing an unparsable folder (like the TV root dir) it will throw an exception, which we want to ignore
-        parse_result = NameParser(try_indexers=True).parse(dirName)
-    except (InvalidNameException, InvalidShowException):  # ignore the exception, because we kind of expected it, but create parse_result anyway so we can perform a check on it.
-        parse_result = False
-
-    search_sql = "SELECT tv_episodes.indexerid, history.resource FROM tv_episodes INNER JOIN history ON history.showid=tv_episodes.showid"  # This part is always the same
-    search_sql += " WHERE history.season=tv_episodes.season AND history.episode=tv_episodes.episode"
-
-    # If we find a showid, a season number, and one or more episode numbers then we need to use those in the query
-    if parse_result and parse_result.show.indexerid and parse_result.episode_numbers and parse_result.season_number:
-        search_sql += " AND tv_episodes.showid=%s AND tv_episodes.season=%s AND tv_episodes.episode=%s" % \
-            (parse_result.show.indexerid, parse_result.season_number, parse_result.episode_numbers[0])
-
-    search_sql += " AND tv_episodes.status IN (" + ",".join([str(x) for x in common.Quality.DOWNLOADED]) + ")"
-    search_sql += " AND history.resource LIKE ? LIMIT 1"
-    sql_result = main_db_con.select(search_sql, ['%' + videofile])
-    if sql_result:
-        # result.output += logHelper(u"You're trying to post process a video that's already been processed, skipping", logger.DEBUG)
+    history_result = main_db_con.select(
+        'SELECT file_size '
+        'FROM tv_episodes e '
```

Comment:
already processed check should be done in history table not in tv_episode. If user processed multiple releases, you will only check the last one by checking tv_episodes table